File: Workspace/Personal Training/JSW/LeetCode/20. Valid Parentheses/Solution.java
class Solution {
    public boolean isValid(String s) {
        char[] charArray = s.toCharArray();
        char[] stack = new char[charArray.length];
        int index = 0;
        for(char c : charArray) {
            if (c == '(' || c == '[' || c == '{') {
                stack[index++] = c;
            }
            else {
                if (index == 0) return false;
                if (stack[index-1] == '(' && c == ')'
                || stack[index-1] == '[' && c == ']'
                || stack[index-1] == '{' && c == '}') {
                    index--;
                }
                else return false;
            }
        }

        return index == 0 ? true : false;
    }
}

File: Workspace/Personal Training/JSW/LeetCode/20. Valid Parentheses/Solution.cs
public class Solution {
    public bool IsValid(string s) {
        char[] stack = new char[s.Length];
        int IDX = 0;
        foreach(char c in s) {
            if (c == '(' || c == '[' || c == '{') stack[IDX++] = c;
            else {
                if (IDX == 0) return false;
                if (!(c == ')' && stack[IDX-1] == '(' ||
                     c == ']' && stack[IDX-1] == '[' ||
                     c == '}' && stack[IDX-1] == '{')) return false;
                
                IDX--;
            }
        }

        return IDX == 0 ? true : false;
    }
}